### PR TITLE
Node.js/week3

### PR DESCRIPTION
**Pull request by hnnhack**

### CODE REVIEW

File: week3/homework/src/todo.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    todo.done = true;
+    await this._save(todos);
+    return todo;
+  }
+
+  async notDone(id) {
+    const todos = await this.read();
+    const todo = todos.find(t => t.id === id);
+    if (todo == null) {
+      const error = new Error(`To-do with ID ${id} does not exist`);
```
Comment (gajduk) [inline week3/homework/src/todo.js]: This code is repeating several times in your homework. Is it possible to eliminate the duplication somehow?

File: week3/homework/src/todo.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      fs.writeFile(this._filename, JSON.stringify(todos, null, 2), error =>
+        error == null ? resolve() : reject(error),
+      );
+    });
+  }
+
+  // under this line: homework assignment 4 more actions
+  async clear() {
+    const todos = await this.read();
+    todos.length = 0;
```
Comment (gajduk) [inline week3/homework/src/todo.js]: interesting way to clear an array 👍

File: week3/homework/src/todo.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    await this._save(todos);
+  }
+
+  async done(id) {
+    const todos = await this.read();
+    const todo = todos.find(t => t.id === id);
+    if (todo == null) {
+      const error = new Error(`To-do with ID ${id} does not exist`);
```
Comment (gajduk) [inline week3/homework/src/todo.js]: correct error handling 👍

File: week3/homework/src/actions/markAsNotDone.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    .then(() => {
+      const id = request.params.id;
+      return todo.notDone(id);
+    })
+    .then(todo => {
+      response.status(200);
+      response.json({ todo });
+    })
+    .catch(({ message, code }) => {
+      response.status(code === 'not-found' ? 404 : 500);
```
Comment (gajduk) [inline week3/homework/src/actions/markAsNotDone.js]: this code is also repeating many times. Can you think of a way to eliminate the duplication?

File: week3/homework/src/todo.js (hunk 1 above)
Comment (hnnhack) [reply week3/homework/src/todo.js]: Yes, you are right. It can possible. I will try to find for eliminate.

File: week3/homework/src/todo.js (hunk 3 above)
Comment (hnnhack) [reply week3/homework/src/todo.js]: Thanks,..

File: week3/homework/src/actions/markAsNotDone.js (hunk 4 above)
Comment (hnnhack) [reply week3/homework/src/actions/markAsNotDone.js]: Ofcourse, I can think about it.

**Comment by hnnhack** [comment]
Hello @gajduk!
Many thanks for your review, great point to recover and your exact source.

**Pull request closed without merging**